### ISSUE TITLE
Pattern json import and export has stopped working on Wordpress.COM

### ISSUE THREAD

**Issue by JungleGenius** [WordPress/gutenberg#68367]
### Description

Hi All,
I'm just a random user of Wordpress.COM but considering this is the last commit to this file and the pattern json import and export was working up until two weeks ago I think this commit may have broken it. On Wordpress.COM I can no longer import pattern json files because I now get the following error "e.getLocationWithParams is not a function". I've tried contacting support over on Wordpress.COM as I have a paid hosting with them but their support is AWOL so I'm dropping this here. I may be wrong and if I am I apologize.

### Step-by-step reproduction instructions

On Wordpress.COM export a pattern json
Then on Wordpress.COM import a pattern json
Now you get the following error "e.getLocationWithParams is not a function".

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by himanshupathak95** [comment]
Hey @JungleGenius, I also got the same error after importing the just exported pattern - 

![1.png](https://github.com/user-attachments/assets/2bc82b09-1dba-43ce-8f43-e3595976440e)



After testing the same on the trunk, I found out that it is actually `history.getLocationWithParams is not a function`

I will first see if there is any issue for this already, and if there isn't I'll start working on the solution.

**Comment by tyxla** [comment]
I can reproduce this on `trunk` and have a fix coming up.

**Comment by tyxla** [comment]
Here's a fix: #68388

**Comment by JungleGenius** [comment]
Hi Folks @ramonjd @tyxla @himanshupathak95,

Thanks you for the swift response and bug fix.  I was trying to get a landing/order page up over the Holliday when I ran headlong into this bug.

I am really impressed with your team's actions once I got the bug got on your radar.  (The front line Wordpress.COM user support was lacking, but it look like the dev team is really on the ball response wise). 

Normally I would just create a pull request for the bug fix like I did for the [Groundhog](https://github.com/groundhoggwp/groundhogg) plug-in over [here](https://github.com/groundhoggwp/groundhogg/pull/65), but I'm not currently setup for WordPress core development (I mostly do Laravel and CodeIgniter).  

The theme/plug-in editor won't let me view or debug the core files so I was kind of stuck.  I ended up pasting the JSON into the pages manually to work around this, but it was buggy because the block editor needs to "resolve" the pasted json code.

I think you are going to need a dot release to get this into Wordpress.COM in the short term.

Thanks everyone.  Great work.

-Daniel
